pkg/object/httpserver/listen.go

```diff
@@ -27,46 +28,51 @@ import (
 // NewLimitListener returns a Listener that accepts at most n simultaneous
 // connections from the provided Listener.
 func NewLimitListener(l net.Listener, n uint32) *LimitListener {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The `context.Background()` will return a none-deadline, forever waiting's context, should we limit this wait time here?

Reply:
It's a canceling function do the similar work as `done`

Reply:
No, we don't need to do this, the `ctx` is only to ensure `sem.Acquire` could be cancelled at the close of the listener.

Reply:
* Yes, the `cancel function does close this context, but one client request may be hanged at the connection stage forever if it doesn't close this connection manually and firstly (If it can't acquire sem successfully due to poor luck).
Although the HTTPServer can eventually close this connection when it shutdown.

* This HTTPSever behavior design can be continued, I will figure it out at another new issue if necessary.